docs/advanced_reference/developer_tooling/benchmarking.md

```diff
@@ -1,23 +1,34 @@
-# Benchmarks
+The benchmarking tool allows measuring the performance of an agent service. The tool is accessed through the class
 
-The benchmarking tools allow measuring the performance of an agent service realised on the {{open_autonomy}} framework.
+```python
+packages.valory.skills.abstract_round_abci.models.BenchmarkTool
+```
 
-## How to use benchmarking tools
+The benchmarking information is saved upon calling the method `BenchmarkTool.save()`. This function call is typically executed within the `async_act()` method of a behaviour. Therefore, the sequence of events that should occur so that the benchmarking information is saved correctly is as follows:
 
-1. Setup agent runtime environment in dev mode using `autonomy deploy build KEYS --dev`.
+1. A behaviour executes the `BenchmarkTool.save()` method.
+2. The agent service reaches the `reset_and_pause` round, and the Tendermint data is dumped.
+3. Stop the service execution after, at least, steps 1. and 2. have occurred in that sequence.
```

Comment:
This is not clear either. Also incorrect I think, which would explain why it is unclear. 
- This  `BenchmarkTool.save()` is currently only used in the `reset_and_pause_abci`. This should be made clear.
https://github.com/valory-xyz/open-autonomy/blob/1227bc5583daf218ac1e1516ac0d0abe22255b18/packages/valory/skills/reset_pause_abci/behaviours.py#L82
- hence the order is different: first ensure the agent enacts the `ResetAndPauseBehaviour`, then the data is saved. 
- This `BenchmarkTool` is unrelated to Tendermint data entirely to my knowledge, why do you reference it here? If I'm wrong please clarify for the next dev to read this how it is related.
- point 3. is not needed is it? Its just confusing to me. I can keep running and save the data each time the agent enacts the `ResetAndPauseBehaviour`; there is no need to stop execution of an agent service to use the `BenchmarkTool` hence I believe this can be removed entirely.

Reply:
ok it is slightly more clear to me now why you mention `Tendermint`, ebcause we have the `consensus` block type.
https://github.com/valory-xyz/open-autonomy/blob/1227bc5583daf218ac1e1516ac0d0abe22255b18/packages/valory/skills/abstract_round_abci/models.py#L701-L705
however all that we save if the `data` 
https://github.com/valory-xyz/open-autonomy/blob/1227bc5583daf218ac1e1516ac0d0abe22255b18/packages/valory/skills/abstract_round_abci/models.py#L756
which consists of timing of execution only
https://github.com/valory-xyz/open-autonomy/blob/1227bc5583daf218ac1e1516ac0d0abe22255b18/packages/valory/skills/abstract_round_abci/models.py#L732-L744

and therefor referencing Tendermint data here is still confusing

Reply:
Fixed - rewritten.